### PR TITLE
Use WebAssembly.instantiateStreaming when presented in browser

### PR DESCRIPTION
**Pull request by chicoxyzzy**
As I proposed in 
https://github.com/JetBrains/kotlin-native/pull/2450#issuecomment-445585728

Still not sure where the tests are and how to run them.

### CODE REVIEW

File: runtime/src/launcher/js/launcher.js [hunk 1]
```diff
@@ -195,11 +195,17 @@ function invokeModule(inst, args) {
 
 // Instantiate module in Browser in a sequence of promises.
 function instantiateAndRun(arraybuffer, args) {
-    WebAssembly.compile(arraybuffer)
-        .then(module => {
-            linkJavaScriptLibraries();
-            return WebAssembly.instantiate(module, konan_dependencies);
-        }).then(instance => invokeModule(instance, args));
+    if (typeof WebAssembly.compileStreaming === 'function') {
+        linkJavaScriptLibraries();
```
Comment (chicoxyzzy) [inline runtime/src/launcher/js/launcher.js]: should it be moved to line before if statement?

**Comment by chicoxyzzy** [comment]
@olonho yes, this instantiates and compiles wasm module using streams so resulting instance will be ready as far as fetch finishes streaming.

**Comment by chicoxyzzy** [comment]
Do you want me to explain these changes in more details?

File: runtime/src/launcher/js/launcher.js [hunk 2]
```diff
@@ -216,9 +216,15 @@ konan.moduleEntry = function (args) {
             throw new Error('Could not find the wasm attribute pointing to the WebAssembly binary.');
         }
         const filename = document.currentScript.getAttribute("wasm");
-        fetch(filename)
-            .then(response => response.arrayBuffer())
-            .then(arraybuffer => instantiateAndRun(arraybuffer, [filename]));
+        if (typeof WebAssembly.instantiateStreaming === 'function') {
+            linkJavaScriptLibraries();
+            WebAssembly.instantiateStreaming(fetch(filename), konan_dependencies)
+              .then(resultObject => invokeModule(resultObject.instance, [filename]));
```
Comment (alexander-gorshenev) [inline runtime/src/launcher/js/launcher.js]: May be extract this scenarion as a third function, similar to instantiateAndRun and instantiateAndRunSync.

Comment (chicoxyzzy) [reply runtime/src/launcher/js/launcher.js]: Should it take only filename as an argument? Like
```js
// Instantiate module in Browser as a promise of streaming instantiation.
function instantiateAndRunStreaming(filename) {
    linkJavaScriptLibraries();
    WebAssembly.instantiateStreaming(fetch(filename), konan_dependencies)
      .then(resultObject => invokeModule(resultObject.instance, [filename]));
}
```

Comment (alexander-gorshenev) [reply runtime/src/launcher/js/launcher.js]: that's okay, I suppose.

**Pull request merged**